### ISSUE TITLE
Pilot coverage for unit tests

### ISSUE THREAD

**Issue by mzabaluev** [movementlabsxyz/movement#407]
Discussion: #396

Add job(s) to CI to collect coverage metrics on the unit tests (i.e. tests ran by `cargo test` in the workspace with no external setup) of all workspace crates that currently exhibit such tests. To get good results, it's suggested to use the `cargo-llvm-cov` tools.

Goals:
* Collect line and region coverage.
* Upload results to analyze, preferably using a code-browsing service like codecov.io.

Stretch goals:
* Get differential reports and notes on coverage degradation.

Non-goals:
* Reach any particular coverage target in process of resolving this issue.

**Comment by l-monninger** [comment]
Sounds good! Because the extent to which this does not include e2e tests is dependent on #393, I will simply regard everything run with `cargo test` currently as valid.

**Comment by mzabaluev** [comment]
You might have to list specific crates for `cargo test`. I think currently the workspace also wants to run e2e tests by default, but feel free to feature-gate that.
